src/dex/uniswap-v2/uniswap-v2.ts

```diff
@@ -340,7 +366,15 @@ export class UniswapV2
     } else {
       pair = { token0, token1, exchange };
     }
-    this.pairs[key] = pair;
+
+    this.dexHelper.cache.setexAndCacheLocally(
```

Comment:
I think it should not cache if `exchange` is NULL_ADDRESS because we already know token0 and token1 and it just consumed cache memory.

Reply:
Same for `findSolidlyPair`.

Reply:
Cache only the `exchange` or its absence as discussed.

Reply:
@skypper updated